### ISSUE TITLE
The swin backbone model trained in train.py cannot be used in test.py  

### ISSUE THREAD

**Issue by AlenLi817** [open-mmlab/mmclassification#839]
### Checklist
- I have searched related issues but cannot get the expected help.
- I have read related documents and don't know what to do.

### Describe the question you meet

I used the tools/train.py and just set config file:
`    parser.add_argument('--config', default=r'configs/swin-tiny_1xb32_zuoCfg_isic2018.py', help='train config file path')`

and got a weight file:
_work_dirs/swin-tiny_1xb32_zuoCfg_isic2018/best_accuracy_top-1_epoch_1.pth_

I tried using this weight file in tools/test.py:
```
    parser.add_argument('--config', default=r'configs/swin-tiny_1xb32_zuoCfg_isic2018.py', help='test config file path')
    parser.add_argument('--checkpoint', default=r'work_dirs/swin-tiny_1xb32_zuoCfg_isic2018/best_accuracy_top-1_epoch_1.pth', help='checkpoint file')
    parser.add_argument('--out',default='pkl_test_results/'
                                        'swin-tiny_1xb32_zuoCfg_isic2018_test_result.pkl', help='output result file')
```

an error:
```
load checkpoint from local path: work_dirs/swin-tiny_1xb32_zuoCfg_isic2018/best_accuracy_top-1_epoch_1.pth
Traceback (most recent call last):
  File "E:/pyproject/mmclassification-master/alentest.py", line 251, in <module>
    main()
  File "E:/pyproject/mmclassification-master/alentest.py", line 174, in main
    checkpoint = load_checkpoint(model, args.checkpoint, map_location='cpu')
  File "C:\anaconda3\lib\site-packages\mmcv\runner\checkpoint.py", line 563, in load_checkpoint
    load_state_dict(model, state_dict, strict, logger)
  File "C:\anaconda3\lib\site-packages\mmcv\runner\checkpoint.py", line 80, in load_state_dict
    load(module)
  File "C:\anaconda3\lib\site-packages\mmcv\runner\checkpoint.py", line 78, in load
    load(child, prefix + name + '.')
  File "C:\anaconda3\lib\site-packages\mmcv\runner\checkpoint.py", line 73, in load
    module._load_from_state_dict(state_dict, prefix, local_metadata, True,
  File "E:\pyproject\mmclassification-master\mmcls\models\backbones\swin_transformer.py", line 456, in _load_from_state_dict
    super()._load_from_state_dict(state_dict, prefix, local_metadata,
  File "C:\anaconda3\lib\site-packages\torch\nn\modules\module.py", line 1372, in _load_from_state_dict
    hook(state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs)
  File "E:\pyproject\mmclassification-master\mmcls\models\backbones\swin_transformer.py", line 514, in _prepare_relative_position_bias_table
    relative_position_bias_table_current = state_dict_model[key]
KeyError: 'backbone.stages.0.blocks.0.attn.w_msa.relative_position_bias_table'

Process finished with exit code 1
```

### Post related information
1. The output of `pip list | grep "mmcv\|mmcls\|^torch"`
```
mmcls                     0.22.1             
mmcv-full                 1.4.2      
```           
2. Your config file if you modified it or created a new one.
swin-tiny_1xb32_zuoCfg_isic2018.py :
```python
dataset_type = 'ISIC2018'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='RandomResizedCrop', size=224),
    dict(type='RandomFlip', flip_prob=0.5, direction='horizontal'),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='ToTensor', keys=['gt_label']),
    dict(type='Collect', keys=['img', 'gt_label'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='Resize', size=(256, -1)),
    dict(type='CenterCrop', crop_size=224),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='Collect', keys=['img'])
]
data = dict(
    samples_per_gpu=32,
    workers_per_gpu=4,
    train=dict(
        type='ISIC2018',
        data_prefix='E:/pyproject/mmclassification-master/data/isic2018/train',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='RandomResizedCrop', size=224),
            dict(type='RandomFlip', flip_prob=0.5, direction='horizontal'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='ToTensor', keys=['gt_label']),
            dict(type='Collect', keys=['img', 'gt_label'])
        ]),
    val=dict(
        type='ISIC2018',
        data_prefix='E:/pyproject/mmclassification-master/data/isic2018/val',
        ann_file=None,
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='Resize', size=(256, -1)),
            dict(type='CenterCrop', crop_size=224),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ]),
    test=dict(
        type='ISIC2018',
        data_prefix='E:/pyproject/mmclassification-master/data/isic2018/test',
        ann_file=None,
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='Resize', size=(256, -1)),
            dict(type='CenterCrop', crop_size=224),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ]))
evaluation = dict(
    interval=1,
    metric=['accuracy'],
    metric_options=dict(topk=(1, )),
    save_best='auto')
checkpoint_config = dict(interval=5)
log_config = dict(interval=100, hooks=[dict(type='TextLoggerHook')])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
model = dict(
    type='ImageClassifier',
    backbone=dict(
        type='SwinTransformer', arch='tiny', img_size=224, drop_path_rate=0.2),
    neck=dict(type='GlobalAveragePooling'),
    head=dict(
        type='LinearClsHead',
        num_classes=7,
        in_channels=768,
        init_cfg=None,
        loss=dict(type='CrossEntropyLoss', loss_weight=1.0),
        cal_acc=False),
    init_cfg=[
        dict(type='TruncNormal', layer='Linear', std=0.02, bias=0.0),
        dict(type='Constant', layer='LayerNorm', val=1.0, bias=0.0)
    ])
paramwise_cfg = dict(
    norm_decay_mult=0.0,
    bias_decay_mult=0.0,
    custom_keys=dict({
        '.absolute_pos_embed': dict(decay_mult=0.0),
        '.relative_position_bias_table': dict(decay_mult=0.0)
    }))
optimizer = dict(
    type='AdamW',
    lr=0.0001,
    weight_decay=0.05,
    eps=1e-08,
    betas=(0.9, 0.999),
    paramwise_cfg=dict(
        norm_decay_mult=0.0,
        bias_decay_mult=0.0,
        custom_keys=dict({
            '.absolute_pos_embed': dict(decay_mult=0.0),
            '.relative_position_bias_table': dict(decay_mult=0.0)
        })))
optimizer_config = dict(grad_clip=None)
lr_config = dict(
    policy='CosineAnnealing',
    by_epoch=True,
    min_lr_ratio=1e-06,
    warmup=None,
    warmup_ratio=0.001,
    warmup_iters=1,
    warmup_by_epoch=True)
runner = dict(type='EpochBasedRunner', max_epochs=100)
work_dir = './work_dirs\swin-tiny_1xb32_zuoCfg_isic2018'
gpu_ids = [0]

```

**Comment by Ezra-Yu** [comment]
try to update your mmcls version.

**Comment by AlenLi817** [comment]
> 
It seems that the latest version of MMCLS has not changed SwinTransformer.  
Is it possible that this problem occurred after v0.22.1 changed SwinTransformer?
### Changelog
**v0.23.0(1/5/2022)**
New Features
Support DenseNet. (https://github.com/open-mmlab/mmclassification/pull/750)
Support VAN. (https://github.com/open-mmlab/mmclassification/pull/739)
Improvements
Support training on IPU and add fine-tuning configs of ViT. (https://github.com/open-mmlab/mmclassification/pull/723)
Docs Update
New style API reference, and easier to use! Welcome [view it](https://mmclassification.readthedocs.io/en/master/api/models.html). (https://github.com/open-mmlab/mmclassification/pull/774)
**v0.22.1(15/4/2022)**
New Features
[Feature] Support resize relative position embedding in SwinTransformer. (https://github.com/open-mmlab/mmclassification/pull/749)
[Feature] Add PoolFormer backbone and checkpoints. (https://github.com/open-mmlab/mmclassification/pull/746)
Improvements
[Enhance] Improve CPE performance by reduce memory copy. (https://github.com/open-mmlab/mmclassification/pull/762)
[Enhance] Add extra dataloader settings in configs. (https://github.com/open-mmlab/mmclassification/pull/752)

**Comment by Ezra-Yu** [comment]
we fix that in the bump PR https://github.com/open-mmlab/mmclassification/pull/809

**Comment by AlenLi817** [comment]
> we fix that in the bump PR #809

got it. Thank you for your prompt answer.
